### DISCUSSION TITLE
Magit not correct work with Emacs 27.2 - duplicate text

### DISCUSSION THREAD

**Discussion by alexei-28** [magit/magit#4373] [Support] [answered]
Linux Mint 20

Emacs 27.2

Magit - 20210430.404

I create text file
![s0](https://user-images.githubusercontent.com/303539/116878910-a57c7b00-ac28-11eb-96e5-8df7c4dbd64c.png)

And now I want to stage and commit this file my magit.

magit-status

![s2](https://user-images.githubusercontent.com/303539/116878962-b75e1e00-ac28-11eb-8d2c-808a22409140.png)

To stage I press S

But I get error:

Error running timer 'auto-revert-buffer': (wrong-type-argument-number-of-marker-p nil)

![s3](https://user-images.githubusercontent.com/303539/116879002-c513a380-ac28-11eb-928e-b073724b4fc8.png)

Then I want to commit. So press c

![s4](https://user-images.githubusercontent.com/303539/116879033-cf35a200-ac28-11eb-9b46-e29906d96940.png)

Input commit message: my commit and press C-c-C-c

After I again open file my.file.txt. As you can see file is correct.

![s5](https://user-images.githubusercontent.com/303539/116879061-d8bf0a00-ac28-11eb-9c4d-32fd79309ddb.png)

Nice. Now I restart Emacs. And again open file my.file.txt. And now text is duplicate.

![s6](https://user-images.githubusercontent.com/303539/116879111-e7a5bc80-ac28-11eb-8eb4-10656875920b.png)

P.S. If I remove

(require 'magit)

from init.el then the problem is gone. No more duplication.

**Comment by tarsius** [comment, marked as answer]
I doubt that this is caused by a bug in Magit.

> Error running timer 'auto-revert-buffer': (wrong-type-argument-number-of-marker-p nil)

This error occurs inside `auto-revert-buffer`. Do `M-x debug-on-error` and then trigger the issue again. You should get a backtrace and if you post that (as text, not a screenshot), then that might help figuring out what's going on there.

> Nice. Now I restart Emacs. And again open file my.file.txt. And now text is duplicate.

Is it really? (You seem to say that not loading `magit`, somehow causes the duplicated line to be removed from the file, which is extremely unlikely to be true.

I suspect that some other package plays into this and that there isn't actually a duplicated line in the *file* but that the line is somehow shown twice in the *buffer*.

Try to reproduce this issue without your configuration and without any third-party packages using `magit-emacs-Q-command` as described at https://magit.vc/manual/magit/Debugging-Tools.html.

(I am on vacation and this probably is it from me on this issue for the next two weeks.)

**Reply by alexei-28** [reply]
M-x toggle-debug-on-error


Debugger entered--Lisp error: (wrong-type-argument number-or-marker-p nil)
  auto-revert-handler()
  #f(compiled-function () #<bytecode 0x158b4e25fbf9>)()
  apply(#f(compiled-function () #<bytecode 0x158b4e25fbf9>) nil)
  auto-revert-buffers()
  auto-revert-mode(1)
  magit-turn-on-auto-revert-mode-if-desired()
  magit-turn-on-auto-revert-mode-if-desired("my.file.txt")
  mapc(magit-turn-on-auto-revert-mode-if-desired ("my.file.txt"))
  magit-stage-untracked(nil)
  magit-stage(nil)
  funcall-interactively(magit-stage nil)
  #<subr call-interactively>(magit-stage nil nil)
  apply(#<subr call-interactively> magit-stage (nil nil))
  call-interactively@ido-cr+-record-current-command(#<subr call-interactively> magit-stage nil nil)
  apply(call-interactively@ido-cr+-record-current-command #<subr call-interactively> (magit-stage nil nil))
  call-interactively(magit-stage nil nil)
  command-execute(magit-stage)


I run Emacs from terminal like this (This shell command starts emacs with only magit and its dependencies loaded):


`/snap/emacs/1090/usr/bin/emacs -Q --eval \(setq\ debug-on-error\ t\) -L /home/alexei/.emacs.d/elpa/dash-20210330.1544/ -L /home/alexei/.emacs.d/elpa/ghub-20210427.1239/ -L /home/alexei/.emacs.d/elpa/graphql-0.1.1/ -L /home/alexei/.emacs.d/elpa/lv-20200507.1518/ -L /home/alexei/.emacs.d/elpa/magit-popup-20200719.1015/ -L /home/alexei/.emacs.d/elpa/transient-20210426.2141/ -L /home/alexei/.emacs.d/elpa/treepy-20191108.2217/ -L /home/alexei/.emacs.d/elpa/with-editor-20210427.1244/ -L /home/alexei/.emacs.d/elpa/magit-2.90.1/ -L /home/alexei/.emacs.d/elpa/git-commit-20210328.1730/ -l /home/alexei/.emacs.d/elpa/magit-2.90.1/magit`


and the problem is gone. 
And what does it mean? That problem is not in magit package?

**Reply by alexei-28** [reply]
I found the  problem.

In my init.el I has :

`'(auto-revert-tail-mode t t)`

If remove this setting then problem (duplication) is gone.

So the  problem is not in magit. 

Thanks

**Reply by alexei-28** [reply]
But I think  this problem appears when use **magit**  AND ` '(auto-revert-tail-mode t t)`.
If use **only**  `(auto-revert-tail-mode t t)` - the problem is gone.
If use **only** package **magit** the problem is gone.
